packages/vitest/src/node/coverage.ts

```diff
@@ -130,12 +131,18 @@ export class BaseCoverageProvider<Options extends ResolvedCoverageOptions<'istan
       this.options.reportsDirectory,
       tempDirectory,
     )
+
+    this.roots = ctx.config.project?.length
+      ? ctx.projects.map(project => project.config.root)
+      : [ctx.config.root]
```

Comment:
Let's add a comment to specify this logic:

```suggestion

    // If --project filter is set pick only roots of resolved projects
    this.roots = ctx.config.project?.length
      ? ctx.projects.map(project => project.config.root)
      : [ctx.config.root]
```